lib/core/src/Cardano/Wallet/Primitive/CoinSelection/MA/RoundRobin.hs

```diff
@@ -168,36 +182,158 @@ data BalanceInsufficientError = BalanceInsufficientError
 --
 performSelection
     :: forall m. (HasCallStack, MonadRandom m)
-    => SelectionCriteria
-    -> m (Either SelectionError SelectionResult)
-performSelection SelectionCriteria {outputsToCover, utxoAvailable}
+    => (TokenMap -> Coin)
+        -- ^ A function which computes the minimum required Ada coins for a
+        -- particular output.
+    -> (SelectionResult [AssetId] -> Coin)
```

Comment:
**Proposal**: change this type to `SelectionResult (Set AssetId)`

**Justification**:

1. The type makes it clear that asset IDs cannot be duplicated in any given change output.
2. Fewer `Set a -> [a]` conversions needed.
3. Similarly, `transformChange` can be simplified to:
    ```hs
    transformChange :: NonEmpty TokenBundle -> NonEmpty (Set AssetId)
    transformChange = fmap TokenBundle.getAssets
    ```